app/lib/common/providers/app_install_check_provider.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:io';
+
+import 'package:appcheck/appcheck.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+enum ExternalApps { whatsApp, telegram, signal }
+
+final isAppInstalledProvider = FutureProvider.family<bool, ExternalApps>(
+  (ref, externalApp) async {
+    return switch (externalApp) {
+      ExternalApps.whatsApp => await checkAppAvailability(
+          Platform.isAndroid ? 'com.whatsapp' : 'whatsapp://',
+        ),
+      ExternalApps.telegram => await checkAppAvailability(
+          Platform.isAndroid ? 'org.telegram.messenger' : 'tg://',
+        ),
+      ExternalApps.signal => await checkAppAvailability(
+          Platform.isAndroid ? 'org.thoughtcrime.securesms' : 'sgnl://',
+        ),
+    };
+  }, // this means we are running
+);
+
+Future<bool> checkAppAvailability(String package) async {
+  try {
+    final app = await AppCheck().checkAvailability(package);
+    return app != null;
+  } catch (e) {
```

Comment:
logging the error? What kind of error are we ever expecting here?

Reply:
If app is not install then `checkAvailability` function will give `PlatformException` so managed in this way.

<img width="789" alt="Screenshot 2024-12-05 at 11 33 36 AM" src="https://github.com/user-attachments/assets/1e3c61c8-957e-44ba-a97c-f05d1371a4f0">

But anyway, as per your suggestion, let me use `isAppInstalled` which is basically doing the same.
<img width="723" alt="Screenshot 2024-12-05 at 11 34 58 AM" src="https://github.com/user-attachments/assets/f885e0fc-17e9-43db-849a-df53439da67f">